edweiss-app/components/core/calendar.tsx

```diff
@@ -1,35 +1,60 @@
 import { Collections, getDocument } from '@/config/firebase';
 import { useAuth } from '@/contexts/auth';
 import { Course } from '@/model/school/courses';
+import * as ScreenOrientation from 'expo-screen-orientation';
 import React, { useEffect, useRef, useState } from 'react';
 import { ScrollView } from 'react-native-gesture-handler';
 import TTouchableOpacity from './containers/TTouchableOpacity';
 import TView from './containers/TView';
-import For from './For';
-
 import { Day } from './Day';
+import For from './For';
 import { getCurrentDay } from './getCurrentDay';
 import { getCurrentTimeInMinutes } from './getCurrentTimeInMinutes';
 import TText from './TText';
 
 const HOUR_BLOCK_HEIGHT = 80;
 const TOTAL_HOURS = 24;
 
+const DAY_NAMES = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];
+const DAY_ORDER = [1, 2, 3, 4, 5, 6, 0];
 
+export const Calendar = ({ courses, type }: { courses: { id: string; data: Course; }[]; type: "week" | "day" | undefined; }) => {
 
-export const Calendar = ({ courses }: { courses: { id: string; data: Course; }[]; }) => {
     const [currentMinutes, setCurrentMinutes] = useState(getCurrentTimeInMinutes());
     const scrollViewRef = useRef<ScrollView>(null);
     const [user, setUser] = useState<any>(null);
     const userId = useAuth().uid;
 
+    const [format, setFormat] = useState(type);
+    useEffect(() => {
+        if (type === undefined) {
+            setFormat("day");
+        }
+    }, [type]);
+    useEffect(() => {
+        const onOrientationChange = (currentOrientation: ScreenOrientation.OrientationChangeEvent) => {
+            const orientationValue = currentOrientation.orientationInfo.orientation;
+            if (type === undefined) {
+                if (orientationValue == 1 || orientationValue == 2) setFormat("day");
+                else setFormat("week");
+            }
+        };
+        const screenOrientationListener =
+            ScreenOrientation.addOrientationChangeListener(onOrientationChange);
+
+        return () => {
+            ScreenOrientation.removeOrientationChangeListener(screenOrientationListener);
+        };
+    }, []);
+
     useEffect(() => {
         const fetchUser = async () => {
             const userData = await getDocument(Collections.users, userId);
             setUser(userData);
         };
         fetchUser();
     }, [userId]);
+
     useEffect(() => {
```

Comment:
I think this use effect could be integrated inside the one that does orientation change listener since they both do not take any parameters